src/nvim/lua/executor.c

```diff
@@ -916,12 +916,22 @@ int nlua_in_fast_event(lua_State *lstate)
   return 1;
 }
 
+static bool viml_func_is_deferred_safe(const char_u *name)
+{
+  const VimLFuncDef *const fdef = find_internal_func((const char *)name);
+  if (fdef) {
+    return fdef->fast;
+  }
+  // Not a vimL function
+  return false;
+}
+
 int nlua_call(lua_State *lstate)
 {
   Error err = ERROR_INIT;
   size_t name_len;
   const char_u *name = (const char_u *)luaL_checklstring(lstate, 1, &name_len);
-  if (!nlua_is_deferred_safe()) {
+  if (!nlua_is_deferred_safe() && !viml_func_is_deferred_safe(name)) {
```

Comment:
`nlua_is_deferred_safe` returns false if _not_ in a fast context.

Reply:
I originally did this, but @bfredl suggested I separated it. Since `nlua_is_deferred_safe` is used in a ton of places in the API functions I think I agree it should be split. This new condition is only relevant to `nvim_call`.